ios/RCTGoogleAnalyticsBridge/RCTGoogleAnalyticsBridge/RCTGoogleAnalyticsBridge.m

```diff
@@ -138,6 +138,14 @@ - (NSDictionary *)constantsToExport
                                                         target:targetUrl] build]];
 }
 
+RCT_EXPORT_METHOD(setCustomDimensionValue:(NSString *)trackerId dimensionIndex:(NSInteger)dimensionIndex dimensionValue:(NSString *)dimensionValue)
+{
+    id<GAITracker> tracker = [[GAI sharedInstance] trackerWithTrackingId:trackerId];
+    
+    [tracker set:[GAIFields customDimensionForIndex:dimensionIndex] value:dimensionValue];
```

Comment:
Ok, I'm on board.

On Fri, May 20, 2016 at 5:27 PM, Christian Brevik <notifications@github.com>
wrote:

> In
> ios/RCTGoogleAnalyticsBridge/RCTGoogleAnalyticsBridge/RCTGoogleAnalyticsBridge.m
> <https://github.com/idehub/react-native-google-analytics-bridge/pull/36#discussion_r64109828>
> :
>
> > @@ -138,6 +138,14 @@ - (NSDictionary *)constantsToExport
> >                                                          target:targetUrl] build]];
> >  }
> >
> > +RCT_EXPORT_METHOD(setCustomDimensionValue:(NSString *)trackerId dimensionIndex:(NSInteger)dimensionIndex dimensionValue:(NSString *)dimensionValue)
> > +{
> > +    id<GAITracker> tracker = [[GAI sharedInstance] trackerWithTrackingId:trackerId];
> > +
> > +    [tracker set:[GAIFields customDimensionForIndex:dimensionIndex] value:dimensionValue];
>
> Your other idea also works, as optional params for trackScreenView and
> trackEvent. But the problem here being if we do that then those functions
> could grow a bit too big.
>
> Say someone wants to implement setCustomMetric later, I don't want that
> to be optional parameters as well on those methods. It seems cleaner to
> implemented a setCustomDimensionWithScreen and/or
> setCustomDimensionWithEventinstead.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/idehub/react-native-google-analytics-bridge/pull/36/files/a31ab6685d6539850a6e7fa07147f735c9f7f8e1#r64109828>
>



-- 
Pete